app/controllers/AlertController.java

```diff
@@ -61,6 +72,89 @@ public AlertController(final Configuration configuration, final AlertRepository
     }
 
     /**
+     * Adds an alert in the alert repository.
+     *
+     * @return Created if the alert was created successfully, an error code otherwise.
+     */
+    public Result add() {
+        Alert alert;
+        try {
+            final models.view.Alert viewAlert = buildViewAlert(request().body());
+            alert = convertToInternalAlert(viewAlert);
+        } catch (final IOException | ConstraintsViolatedException | IllegalArgumentException e) {
+            LOGGER.error()
+                    .setMessage("Failed to build an alert.")
+                    .setThrowable(e)
+                    .log();
+            return badRequest("Invalid request body.");
+        }
+
+        if (_alertRepository.get(alert.getId()).isPresent()) {
+            LOGGER.error()
+                    .setMessage(String.format("An alert with id %s already exists", alert.getId().toString()))
+                    .log();
+            return new StatusHeader(play.core.j.JavaResults.Conflict());
+        }
+
+        try {
+            _alertRepository.addOrUpdateAlert(alert);
+            // CHECKSTYLE.OFF: IllegalCatch - Convert any exception to 500
+        } catch (final Exception e) {
+            // CHECKSTYLE.ON: IllegalCatch
+            LOGGER.error()
+                    .setMessage("Failed to add an alert.")
+                    .setThrowable(e)
+                    .log();
+            return internalServerError();
+        }
+        return created();
+    }
+
+    /**
+     * Updates an alert in the alert repository.
+     * If a valid UUID is passed, it will create an alert with this id.
+     *
+     * @param alertId The <code>String</code> representation of the id of the alert to be updated.
+     * @return Success if the alert was updated successfully, an error code otherwise.
+     */
+    public Result update(final String alertId) {
+        Alert alert;
+        try {
+            final models.view.Alert viewAlert = buildViewAlert(request().body());
+            if (!alertId.equals(viewAlert.getId())) {
+                return badRequest("alert id in the request body must match the resource id in path");
+            }
+            alert = convertToInternalAlert(viewAlert);
+        } catch (final IOException | ConstraintsViolatedException | IllegalArgumentException e) {
+            LOGGER.error()
+                    .setMessage("Failed to build an alert.")
+                    .setThrowable(e)
+                    .log();
+            return badRequest("Invalid request body.");
+        }
+
+        if (!_alertRepository.get(alert.getId()).isPresent()) {
```

Comment:
To resolve the deletion in addition to the version described above we will need to add a tombstone (aka soft delete) semantic to the data. I assume you would prefer to do these separately. Please file issues for each against the project.